dev/TeachingTip/TeachingTip.cpp

```diff
@@ -971,27 +975,58 @@ void TeachingTip::OnF6AcceleratorKeyClicked(const winrt::CoreDispatcher&, const
             m_hasF6BeenInvoked = true;
         }
 
-        winrt::Button f6Button = [this]() -> winrt::Button
+        auto const focusFromContent = [this, args]()
         {
-            auto firstButton = m_closeButton.get();
-            auto secondButton = m_alternateCloseButton.get();
-            if (CloseButtonContent())
+            auto parent = winrt::FocusManager::GetFocusedElement();
+            auto const rootElement = m_rootElement.get();
+            while (parent)
             {
-                std::swap(firstButton, secondButton);
+                if (auto parentAsUIElement = parent.try_as<winrt::UIElement>())
+                {
+                    if (parentAsUIElement == rootElement)
+                    {
+                        return true;
+                    }
+                }
+                parent = winrt::VisualTreeHelper::GetParent(parent.try_as<winrt::DependencyObject>());
```

Comment:
@Austin-Lamb  Is there no API to ask whether focus is in a subtree? It feels rude to walk the parents and fault in peers.

Reply:
I don't see an API but I *think* we can track whether or not we have focus by watching UIElement.GotFocus / LostFocus on the root element.

Reply:
This was the approach austin suggested. if there is a better way that would be awesome, casting the whole visual tree to get this feature didn't feel great.

Reply:
GotFocus/LostFocus can be more challenging than you'd expect, since both are fired async.  By the time you're handling GotFocus, you may no longer have focus - similarly by the time you handle LostFocus you may not still have focus.  So they can be used, but be careful.

GettingFocus/LosingFocus can sometimes help here, as they're synchronous - but due to the ability to redirect focus during the focus movement operation, they again require careful handling.

---
In reply to: [270645425](https://github.com/Microsoft/microsoft-ui-xaml/pull/511#discussion_r270645425) [](ancestors = 270645425)

Reply:
Unfortunately no, we don't have an equivalent of WPF's IsKeyboardFocusWithin, you do need to walk the tree like this.

---
In reply to: [270632897](https://github.com/Microsoft/microsoft-ui-xaml/pull/511#discussion_r270632897) [](ancestors = 270632897)

Reply:
How about GotFocus/LostFocus? Will they be properly paired even with focus re-direction?

---
In reply to: [270928960](https://github.com/Microsoft/microsoft-ui-xaml/pull/511#discussion_r270928960) [](ancestors = 270928960,270645425)

Reply:
GotFocus/LostFocus will always fire in pairs since they are the "after the fact" notification - getting/losing can fire multiple times during redirection as none of them is guaranteed to be 'final' - anybody can listen after you and redirect, resulting in more getting/losing notifications until eventually it stops and you get a Got/Lost.